src/library/scala/math/BigInt.scala

```diff
@@ -47,8 +55,7 @@ object BigInt {
    *  @return  the constructed `BigInt`
    */
   def apply(l: Long): BigInt =
-    if (minCached <= l && l <= maxCached) apply(l.toInt)
-    else new BigInt(BigInteger.valueOf(l))
+    if (minCached <= l && l <= maxCached) getCached(l.toInt) else new BigInt(l)
```

Comment:
does it pay to do `val i = l.toInt` first? I don't know if the comparisons are converting to int twice behind the scenes.

Reply:
It's more that the minCached, maxCached will be converted to Long. Truncating the Long and then doing the comparison would lead to incorrect results -- or we'd need to check `l.isValidInt`.